src/pepr/operator/controllers/exemptions/exemptions.ts

```diff
@@ -0,0 +1,87 @@
+import { Log } from "pepr";
+import { policies } from "../../../policies/index";
+import { Matcher, Policy, UDSExemption } from "../../crd";
+
+// Remove leading and trailing '/' if added by user to matcher name
+function removeRegexSlash(name: string) {
+  if (name[0] === "/" && name[name.length - 1] === "/") {
+    name = name.slice(1, name.length - 1);
+  }
+  return name;
+}
+
+function isAlreadyAdded(matchers: Matcher[], name: string) {
+  for (const m of matchers) {
+    if (m.name === name) {
+      return true;
+    }
+  }
+}
+
+// Add Exemptions to Pepr store as "policy": "[{matcher}]"
+export function processExemptions(exempt: UDSExemption) {
+  const { Store } = policies;
+  const policyList = Object.values(Policy);
+  const exemptions = exempt.spec?.exemptions ?? [];
+
+  // Use local map for matchers aggregation before writing to store
+  const exemptionMap = new Map<Policy, Matcher[]>();
+
+  // Iterate through all policies -- important for removing exemptions if CR is updated
+  for (const p of policyList) {
+    // Set local map with current state of store
+    exemptionMap.set(p, JSON.parse(Store.getItem(p) || "[]"));
+
+    for (const e of exemptions) {
+      const matchers = exemptionMap.get(p) ?? [];
+      const name = removeRegexSlash(e.matcher.name);
+
+      if (e.policies.includes(p)) {
+        // Do additional checks if policy already has matchers
+        if (matchers.length > 0) {
+          if (isAlreadyAdded(matchers, name)) {
+            continue;
+          } else {
+            matchers.push({ namespace: e.matcher.namespace, name: name });
+            exemptionMap.set(p, matchers);
+          }
+        } else {
+          // Else add to policy for the first time
+          exemptionMap.set(p, [{ namespace: e.matcher.namespace, name: name }]);
+        }
+      } else {
+        // check if policy has this matcher from previous version of CR
+        for (const m of matchers) {
+          if (m.name === name) {
+            Log.debug(`Removing ${name} from ${p}`);
+            exemptionMap.set(
+              p,
+              matchers.filter(m => m.name !== name),
+            );
+          }
```

Comment:
I think this check for removal misses a case where you delete an entire matcher (rather than just a policy under a matcher). 

Probably a broader question/discussion on how we architect removals...need to have some way to get context of previous state/tie a matcher back to the `exemption` CR it came from.